pkg/build/builder/daemonless_test.go

```diff
@@ -2,12 +2,118 @@ package builder
 
 import (
 	"os"
+	"reflect"
 	"strings"
 	"testing"
 
+	"k8s.io/kubernetes/pkg/credentialprovider"
+
 	builderutil "github.com/openshift/builder/pkg/build/builder/util"
 )
 
+func Test_mergeNodeCredentials(t *testing.T) {
+	for _, tt := range []struct {
+		name      string
+		nsCreds   string
+		nodeCreds string
+		errstr    string
+		expected  credentialprovider.DockerConfig
+	}{
+		{
+			name:   "invalid namespace credentials file path",
+			errstr: "no such file or directory",
+		},
+		{
+			name:    "invalid namespace credentials file",
+			nsCreds: "testdata/empty.txt",
+			errstr:  "unexpected end of JSON input",
+		},
+		{
+			name:     "empty namespace credentials",
+			nsCreds:  "testdata/credentials-empty.json",
+			expected: map[string]credentialprovider.DockerConfigEntry{},
+		},
+		{
+			name:    "valid namespace credentials",
+			nsCreds: "testdata/credentials-quayio-user0.json",
+			expected: map[string]credentialprovider.DockerConfigEntry{
+				"quay.io": {
+					Username: "user0",
+					Password: "pass0",
+					Email:    "user0@redhat.com",
+				},
+			},
+		},
+		{
+			name:      "merge namespace with node credentials",
+			nsCreds:   "testdata/credentials-quayio-user0.json",
+			nodeCreds: "testdata/credentials-redhatio-nodeuser.json",
+			expected: map[string]credentialprovider.DockerConfigEntry{
+				"quay.io": {
+					Username: "user0",
+					Password: "pass0",
+					Email:    "user0@redhat.com",
+				},
+				"registry.redhat.io": {
+					Username: "nodeuser",
+					Password: "nodepass",
+					Email:    "nodeuser@redhat.com",
+				},
+			},
+		},
+		{
+			name:      "overwriting node credentials",
+			nodeCreds: "testdata/credentials-redhatio-nodeuser.json",
+			nsCreds:   "testdata/credentials-redhatio-nsuser.json",
+			expected: map[string]credentialprovider.DockerConfigEntry{
+				"registry.redhat.io": {
+					Username: "nsuser",
+					Password: "nspass",
+					Email:    "nsuser@redhat.com",
+				},
+			},
+		},
+		{
+			name:      "invalid node credentials",
+			nsCreds:   "testdata/credentials-quayio-user0.json",
+			nodeCreds: "testdata/empty.txt",
```

Comment:
@ricardomaraschini This file does not exist - was this intentional?

Reply:
It [does exist](https://github.com/openshift/builder/blob/master/pkg/build/builder/testdata/empty.txt), it is not part of this PR though.